Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -185,63 +186,77 @@ section VariableChange
 
 /-! ### Variable changes -/
 
-variable (u : Rˣ) (r s t : R)
+/-- An admissible linear change of variable of Weierstrass curves defined over a ring `R`.
+It consists of a tuple $(u,r,s,t)$ of elements in $R$, with $u$ invertible.
+As a matrix, it is $\begin{pmatrix}
+u^2 & 0 & r \cr
+u^2s & u^3 & t \cr
+0 & 0 & 1
+\end{pmatrix}$. -/
+@[ext]
+structure VariableChange (R : Type u) [CommRing R] where
+  (u : Rˣ) (r s t : R)
+
+variable (C : VariableChange R)
 
 /-- The Weierstrass curve over `R` induced by an admissible linear change of variables
 $(X, Y) \mapsto (u^2X + r, u^3Y + u^2sX + t)$ for some $u \in R^\times$ and some $r, s, t \in R$. -/
 @[simps]
 def variableChange : WeierstrassCurve R where
-  a₁ := ↑u⁻¹ * (W.a₁ + 2 * s)
-  a₂ := ↑u⁻¹ ^ 2 * (W.a₂ - s * W.a₁ + 3 * r - s ^ 2)
-  a₃ := ↑u⁻¹ ^ 3 * (W.a₃ + r * W.a₁ + 2 * t)
-  a₄ := ↑u⁻¹ ^ 4 * (W.a₄ - s * W.a₃ + 2 * r * W.a₂ - (t + r * s) * W.a₁ + 3 * r ^ 2 - 2 * s * t)
-  a₆ := ↑u⁻¹ ^ 6 * (W.a₆ + r * W.a₄ + r ^ 2 * W.a₂ + r ^ 3 - t * W.a₃ - t ^ 2 - r * t * W.a₁)
+  a₁ := ↑C.u⁻¹ * (W.a₁ + 2 * C.s)
+  a₂ := ↑C.u⁻¹ ^ 2 * (W.a₂ - C.s * W.a₁ + 3 * C.r - C.s ^ 2)
+  a₃ := ↑C.u⁻¹ ^ 3 * (W.a₃ + C.r * W.a₁ + 2 * C.t)
+  a₄ := ↑C.u⁻¹ ^ 4 * (W.a₄ - C.s * W.a₃ + 2 * C.r * W.a₂
+    - (C.t + C.r * C.s) * W.a₁ + 3 * C.r ^ 2 - 2 * C.s * C.t)
+  a₆ := ↑C.u⁻¹ ^ 6 * (W.a₆ + C.r * W.a₄ + C.r ^ 2 * W.a₂
+    + C.r ^ 3 - C.t * W.a₃ - C.t ^ 2 - C.r * C.t * W.a₁)
```

Comment:
My convention is to have as many terms as possible in the same line, so e.g. `- (C.t + C.r * C.s) * W.a₁` (and maybe even `+ 3 * C.r ^ 2`) should be in the same line, and then line break afterwards